tldw_Server_API/app/core/Ingestion_Media_Processing/Audio/Audio_Transcription_Lib.py

```
@@ -2841,6 +2911,22 @@ class ConversionError(Exception):
     pass
 
+
+def _check_cancel(cancel_check: Optional[Callable[[], bool]], *, label: str) -> None:
+    if cancel_check is None:
+        return
+    try:
+        should_cancel = cancel_check()
+        if inspect.isawaitable(should_cancel):
+            should_cancel = asyncio.run(should_cancel)
+        if should_cancel:
+            raise TranscriptionCancelled(f"Cancelled during {label}")
```

Ruff (0.14.13)

Abstract `raise` to an inner function

(TRY301)